cluster/gce/gci/configure-helper.sh

```diff
@@ -1299,7 +1299,13 @@ function prepare-kube-proxy-manifest-variables {
     params+=" --feature-gates=${FEATURE_GATES}"
   fi
   if [[ "${KUBE_PROXY_MODE:-}" == "ipvs" ]];then
-    params+=" --proxy-mode=ipvs --feature-gates=SupportIPVSProxyMode=true"
+    sudo modprobe -a ip_vs ip_vs_rr ip_vs_wrr ip_vs_sh nf_conntrack_ipv4
+    if [[ $? -eq 0 ]];
```

Comment:
If this fails on some nodes but not others, then we end up with a cluster in an inconsistent that that isn't detectible. Would it be better to just fail outright if this doesn't work? Then we would find out pretty quickly how reliable the above command is.

Reply:
Would "exit 1" be the appropriate way to fail?

Reply:
I'll test this against a GKE cluster and see what happens. Since COS does not have those modules yet, the loading should fail so I can see what the behavior is like.

Reply:
Now that I think about it more, I'm not sure the best practice here would be to fail. The standard way of doing things in kube-proxy (in the code itself) is to gracefully fall back to a more usable mode if the prerequisites for the specified mode are not present. For example, if ipvs is specified but the modules are not there, it will automatically fall back to iptables. If the required stuff for iptables is not there, it will fall back to userspace. For the sake of keeping with the standard, I think it would be better in the scripts to just fall back to another mode rather than fail.  Regarding your point about it failing on some nodes but not others, this looks it already has the potential for happening since kube-proxy in not failing itself.

Reply:
Graceful degradation is great, and I see your argument for applying it here. But if a system operator (or automation) can't tell that it has happened, then they have no way to decide whether they want to make a remediation to the system to fix it and get the system back into a non-degraded mode. 

Is the fallback from iptables --> user mode discoverable in any way?

@thockin @bowei

Reply:
AFAIK, kube-proxy already have this degradation logic, to check whether kernel modules loaded or not, if not, print log and fall back. I think we don't have to fall back in this script.
@m1093782566 is this make sense?

Reply:
> But if a system operator (or automation) can't tell that it has happened, then they have no way to decide whether they want to make a remediation to the system to fix it and get the system back into a non-degraded mode.

+1

Why not just "exit 1" to make every node's behavior more consistent?

Reply:
@Lion-Wei I'm not sure I understand your comment. Isn't the goal to remove all kernel module logic from kube-proxy itself? Therefore, if we decide to fall back or exit, that logic would be reflected in the script.

@m1093782566 I'm fine with exiting, just wanted the explore the option :)